### PR TITLE
Validate date of birth

### LINKED ISSUE #127: date of birth should be less than date.now

### PR DESCRIPTION
**Pull request by sandeshsoni**
# Fixes issue number
fixes #127 

# Describe the changes made on this PR
- add a validation, date-of-birth can be same date or past date only
-
-
-

# Visuals
If your merge request had visual changes we need the screenshot of the affected pages and if behaviour changed
we will need a gif showing the new features

## Screenshots
Screenshot 2019-04-01 21 53 43

### Mobile view

Attach screenshot of changes as they appear on a small screen

### Desktop view
Attach screenshot of the changes as they appear on larger screens

## GIF
If there were changes on how the application behaves then attach a gif showing the new changes

# Reviews
- [x] I have formated by code as expected
We have a handy tool that helps you check if your elixir code is well formatted. If the tool passes on your local build
then it will definitly pass our linting checks online

- [x] I have filled in all the required information above
- [x] I have marked my work as `ready for review`

### CODE REVIEW

**Comment by sandeshsoni** [comment]
@GettyOrawo Please confirm if a child born today can be added or not.
I used my judgement and added logic. Correct me if the logic looks wrong to you.

**Comment by sigu** [comment]
@sandeshsoni yes a child born today can be added on the system

File: lib/nailinda/accounts/patient.ex
Comment (sigu) [inline lib/nailinda/accounts/patient.ex]: Though not strict requirement, we like having private functions start with `_` .
This allows us to know whether its a private function defined within the module or its been imported from other modules

In this case we can have 
``` elixir 
defp _validate_date_is_not_future(changeset) do
......
end
```

Comment (sandeshsoni) [reply lib/nailinda/accounts/patient.ex]: I have updated the PR.

use _ prefix for private function
squashed commits to a single commit.

Comment (GettyOrawo) [reply lib/nailinda/accounts/patient.ex]: This is great @sandeshsoni :smiley:  I love your approach towards this. Will merge it in after the other approvals. Good job!

**Comment by GettyOrawo** [comment]
This is great @sandeshsoni :smiley:  I love your approach towards this. Will merge it in after the other approvals. Good job!

**Review by 75pollet — APPROVED** [review]

**Comment by sigu** [comment]
Lets wait for the conference demonstration by @LindaKadz  before merging this in :-)

**Comment by jackline-kaunda** [comment]
It seems your branch does not have the latest changes from the base branch. I would request you to update your branch (pull from develop) and push  so that we can merge your work.

**Pull request merged**
